### PR TITLE
lscc as service

### PR DESCRIPTION
**Pull request by vitiko**

### CODE REVIEW

File: api/public.go [hunk 1]
```diff
@@ -9,14 +9,19 @@ import (
 	"github.com/hyperledger/fabric/msp"
 )
 
+type CurrentIdentity interface {
+	// CurrentIdentity identity returns current signing identity used by core
+	CurrentIdentity() msp.SigningIdentity
```
Comment (uraldav) [inline api/public.go:14]: just  Identity() ?

File: examples/cli/main.go [hunk 2]
```diff
@@ -32,34 +27,36 @@ var (
...
+	//	ctx,
+	//	*channel,
+	//	*ccPath,
+	//	*ccVersion,
+	//	*ccPolicy,
+	//	util.ToChaincodeArgs(*ccArgs),
+	//	prepareTransArgs(*ccTransient),
+	//); err != nil {
+	//	log.Fatalln(err)
+	//}
```
Comment (uraldav) [inline examples/cli/main.go:59]: uncomment and update along with new code.

Comment (vitiko) [reply examples/cli/main.go:59]: chaincode build with lscc and lifecycle will be in next commit

File: api/public.go (hunk 1 above)
Comment (vitiko) [reply api/public.go:14]: make sense, wil be in next commit

File: client/chaincode/invoke_builder.go [hunk 3]
```diff
@@ -105,31 +99,38 @@ func (b *invokeBuilder) Do(ctx context.Context, options ...api.DoOption) (*fabri
...
 	}
 
-	if err = doOpts.TxWaiter.Wait(ctx, b.ccCore.channelName, tx); err != nil {
-		return nil, tx, err
+	if err = doOpts.TxWaiter.Wait(ctx, b.ccCore.channelName, txID); err != nil {
+		return nil, txID, err
 	}
 
-	return peerResponses[0].Response, tx, nil
+	return peerResponses[0].Response, txID, nil
```
Comment (bogatyr285) [inline client/chaincode/invoke_builder.go:133]: а мы тут точно нулевой индекс иметь будем?

Comment (vitiko) [reply client/chaincode/invoke_builder.go:141]: fixed

**Pull request merged**